Subject: [PATCH] net sched: minor fix for tcf_refcnt
From 825090b5dd26765c0bd68157a7b626feebc2f4b8
From: Fedor <fedork@ubuntu.(none)>
Date: Sat, 22 May 2010 13:58:23 -0400

backport of kernel.org commit c3bc7cff8fddb6ff9715be8bfc3d911378c4d69d

---
 include/net/act_api.h   |   10 +++++++---
 1 files changed, 7 insertions(+), 3 deletions(-)

diff --git a/include/net/act_api.h b/include/net/act_api.h
index 8b06c2f..406d433 100644
--- a/include/net/act_api.h
+++ b/include/net/act_api.h
@@ -146,9 +146,13 @@ tcf_police_release(struct tcf_police *p, int bind)
 		}
 	}
 #else
-	if (p && --p->tcf_refcnt == 0)
-		tcf_police_destroy(p);
-
+	if (p) {
+		p->tcf_refcnt--;
+		if (p->tcf_refcnt <= 0) {
+			tcf_police_destroy(p);
+			ret = 1;
+		}
+	}
 #endif /* CONFIG_NET_CLS_ACT */
 	return ret;
 }
-- 
1.6.5.GIT

